pandas/core/algorithms.py

```diff
@@ -204,6 +204,10 @@ def value_counts(values, sort=True, ascending=False, normalize=False,
     is_period = com.is_period_arraylike(values)
     values = Series(values).values
     is_category = com.is_categorical_dtype(values.dtype)
+    if is_category and dropna:
+        values = values.dropna()
+        if np.nan in values.categories:
```

Comment:
you shouldn't try to do ``np.nan`` in anything....its not generally a safe operation. the ``.dropna()`` should handle this completely.